### PR TITLE
Add some improvements for slight performance gains

### PR DESCRIPTION
**Pull request by yigitksl**
Added a bunch of preprocessing steps. I haven't done what you asked about profanity words yet. Also updated the training loop with a custom loss function that inverse-proportionally punishes the model depending on the weight of the label.

### CODE REVIEW

**Comment by yigitksl** [comment]
reopening
